File: MtxVec.Delphi/Find_Form.pas
unit Find_Form;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls;

type
  TfrmFind = class(TForm)
    Label1: TLabel;
    Edit1: TEdit;
    Button1: TButton;
    Button2: TButton;
    CheckBox1: TCheckBox;
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
  private
    { Private declarations }
  public
    { Public declarations }
    FindAll: boolean;
    FindText: String;
  end;

var
  frmFind: TfrmFind;

implementation

{$IFDEF CLR}
{$R *.NFM}
{$ELSE}
{$R *.DFM}
{$ENDIF}

procedure TfrmFind.FormClose(Sender: TObject; var Action: TCloseAction);
begin
  FindText := Edit1.Text;
  FindAll := CheckBox1.Checked;
end;

end.

File: MtxVec.CPP/Find_Form.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Find_Form.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TfrmFind *frmFind;
//---------------------------------------------------------------------------
__fastcall TfrmFind::TfrmFind(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmFind::FormClose(TObject *Sender, TCloseAction &Action)
{
  FindText = Edit1->Text;
  FindAll = CheckBox1->Checked;
}
//---------------------------------------------------------------------------
